rbac/core/kafka_consumer.py

```diff
@@ -0,0 +1,674 @@
+            return False
+
+        # Future: Add network/gRPC exception types here when gRPC calls are added
+        # if isinstance(exception, grpc.RpcError):
+        #     return True
+        # if isinstance(exception, requests.exceptions.RequestException):
+        #     return True
+
+        # For now, retry connection, timeout errors, and generic ValueErrors (but not validation-specific ones)
+        # This will be expanded when network calls are added to _process_relations_message
+        return isinstance(exception, (ConnectionError, TimeoutError, OSError, ValueError))
+
+    def _process_message_with_retry(
+        self, message_value: Dict[str, Any], message_offset: int, message_partition: int
+    ) -> bool:
+        """Process a message with selective retry logic - only retry network/processing errors."""
+        attempt = 0
+        start_time = time.time()
+
+        while True:
+            try:
+                # Attempt to process the message
+                success = self._process_debezium_message(message_value)
+
+                if success:
+                    if attempt > 0:
+                        total_duration = time.time() - start_time
+                        logger.info(
+                            f"Message successfully processed after {attempt + 1} attempts "
+                            f"(partition: {message_partition}, offset: {message_offset}, "
+                            f"total retry time: {total_duration:.2f}s)"
+                        )
+                        message_retry_duration.labels(retry_reason="processing_error").observe(total_duration)
+
+                    return True
+
+                # If processing failed, this could be validation or business logic failure
+                # Don't retry - validation failures and business logic errors are permanent
+                logger.error(
+                    f"Message processing failed (partition: {message_partition}, "
+                    f"offset: {message_offset}). Message will be skipped.\n"
+                    f"Message content: {message_value}"
+                )
+                messages_processed_total.labels(message_type="unknown", status="processing_failed").inc()
+                return False
+
+            except json.JSONDecodeError as e:
+                # JSON decode errors are permanent - should not retry
+                logger.error(
+                    f"Permanent JSON decode error for message (partition: {message_partition}, "
+                    f"offset: {message_offset}): {e}. Message will be skipped.\n"
+                    f"Message content: {message_value}"
+                )
+                messages_processed_total.labels(message_type="unknown", status="json_error").inc()
+                return False
+
+            except Exception as e:
+                # Only retry specific network/processing exceptions
+                if self._should_retry_exception(e):
```

Comment:
Maybe we can backoff the retry cadence here like 2s for the second time, 4 seconds for the third time etc., so it won't be retrying to soon like 100 times retries within a minute due to network issue.
But this could be a future enhancement

Reply:
yes @astrozzc you are right and it is already here in `calculate_delay` below, 

- it uses formula:
```
delay = self.initial_delay * (self.backoff_multiplier**attempt)
```